src/pydvl/value/oob/oob.py

```diff
@@ -0,0 +1,136 @@
+
+    There is a need to tune n_est and max_samples jointly to ensure all samples are at least 1 time oob, otherwise the result could include a nan value for that datum.
+
+    Args:
+        u: Utility object with model, data, and scoring function.
+        n_est: Number of estimator used in the bagging procedure.
+        max_samples: The fraction of samples to draw to train each base estimator.
+        n_jobs: The number of jobs to run in parallel used in the bagging
+            procedure for both fit and predict.
+        loss: A function taking as parameters model prediction and corresponding
+            data labels(preds, y) and returning an array of point-wise errors.
+        progress: If True, display a progress bar.
+
+    Returns:
+        Object with the data values.
+    """
+
+    result: ValuationResult[np.int_, np.float_] = ValuationResult.empty(
+        algorithm="data_oob", indices=u.data.indices, data_names=u.data.data_names
+    )
+
+    if is_classifier(u.model):
+        bag = BaggingClassifier(
+            u.model, n_estimators=n_est, max_samples=max_samples, n_jobs=n_jobs
+        )
+        if loss is None:
+            loss = point_wise_accuracy
+    elif is_regressor(u.model):
+        bag = BaggingRegressor(
+            u.model, n_estimators=n_est, max_samples=max_samples, n_jobs=n_jobs
+        )
+        if loss is None:
+            loss = neg_l2_distance
+    else:
+        raise Exception(
+            "Model has to be a classifier or a regressor in sklearn format."
+        )
+
+    bag.fit(u.data.x_train, u.data.y_train)
+
+    for est, samples in maybe_progress(
+        zip(bag.estimators_, bag.estimators_samples_), progress, total=n_est
+    ):  # The bottleneck is the bag fitting not this part so TQDM is not very useful here
+        oob_idx = np.setxor1d(u.data.indices, np.unique(samples))
+        array_loss = loss(
+            preds=est.predict(u.data.x_train[oob_idx]), y=u.data.y_train[oob_idx]
+        )
+        result += ValuationResult(
+            algorithm="data_oob",
+            indices=oob_idx,
+            values=array_loss,
+            counts=np.ones_like(array_loss, dtype=u.data.indices.dtype),
+        )
+    return result
+
+
+def point_wise_accuracy(preds: NDArray, y: NDArray) -> NDArray:
+    r"""Computes point wise accuracy
+
+    Args:
+        preds: Model prediction on
+        y:  data labels corresponding to the model predictions
+
+    Returns:
+        Array of point wise accuracy
+    """
+    return np.array(preds == y, dtype=np.int_)
+
+
+def neg_l2_distance(preds: NDArray, y: NDArray) -> NDArray[np.float64]:
+    r"""Computes negative l2 distance between label and model prediction
+
+    Args:
+        preds: Model prediction on
+        y:  data labels corresponding to the model predictions
+
+    Returns:
+        Array with point wise negative l2 distance between label and model prediction
```

Comment:
Since you call `np.square` before returning the array then all the values are positive. Why is it called *negative* l2 distance?

Reply:
A mistake on my part. Thanks Anes. FYI the negative part is there to match the classification values order. The higher the value, the better the result.